src/main/scala/org/apache/spark/sql/delta/commands/UpdateCommand.scala

```diff
@@ -129,10 +135,12 @@ case class UpdateCommand(
       }.asNondeterministic()
       val filesToRewrite =
         withStatusCode("DELTA", s"Finding files to rewrite for UPDATE operation") {
-          data.filter(new Column(updateCondition))
-            .filter(updatedRowUdf())
-            .select(input_file_name())
-            .distinct().as[String].collect()
+          withJobDescription("Filtering files for update", sparkSession) {
```

Comment:
Actually now that I realize `withStatusCode`, shall we just merge `withStatusCode` and `withJobDescription`?

Reply:
They don't always match 1-1
there are at least some places where `withStatusCode` is not called, eg.
`PartitionFiltering`
`  protected lazy val computedState: State =` in `Snapshot`
And other places like
`GenerateSymlinkManifestImpl.recordManifestGeneration` where `withStatusCode` is "far" from Spark query.

But mainly, I think this may cause calls to nest, which is not desirable for `withJobDescription`

Reply:
@brkyvz do think this requires anything else?

Reply:
Yeah, I just think that having one mechanism to state what's going on is actually pretty useful. Otherwise developers may be confused when to use `withJobDescription` versus `withJobDescription`. I feel it'd be a good idea to merge the two. If we do merge the two, it'd be great to just add the `withStatusCode` to places where you've already added the `withJobDescription` (if they don't already exist)